app/controllers/api/subcollections/vms.rb

```diff
@@ -23,16 +23,31 @@ def vms_query_resource(object)
 
       private
 
+      def decorator_selection
+        params['decorators'].to_s.split(",")
+      end
+
+      def decorator_selection_for(collection)
+        decorator_selection.collect do |attr|
+          /\A#{collection}\.(?<name>.*)\z/.match(attr) { |m| m[:name] }
+        end.compact
+      end
+
       def create_vm_attributes_hash(vm_attrs, vm)
         vm_attrs.each_with_object({}) do |attr, hash|
           hash[attr] = vm.public_send(attr.to_sym) if vm.respond_to?(attr.to_sym)
         end.compact
       end
 
       def create_vm_decorators_hash(vm_decorators, vm)
-        vm_decorators.each_with_object({}) do |name, hash|
-          hash[name] = vm.decorate.public_send(name.to_sym) if vm.decorate.respond_to?(name.to_sym)
-        end.compact
+        hash = {}
+        if vm_decorators.include? 'supports_console?'
+          hash['supports_console?'] = vm.supports_console?
+        end
+        if vm_decorators.include? 'supports_cockpit?'
+          hash['supports_cockpit?'] = vm.supports_launch_cockpit?
+        end
+        hash
```

Comment:
may end up with lotsa churn with this in API every-time there's a new decorator. Would prefer logic to be driven by "support_" prefix and calling the appropriate method on vm.  Not sure about the differently named _cockpit? vs. launch_cockpit?, maybe define an alias in the model for compatibility so this code here is generic for all.  Thanks.